core/cog-view.h

```diff
@@ -32,11 +33,22 @@ struct _CogViewClass {
 
 #define COG_TYPE_VIEW_IMPL (cog_view_get_impl_type())
 
+COG_API
 GType                    cog_view_get_impl_type(void);
```

Comment:
Should be this part of the public API? AFAIK this should be used by internals like the cog-shell

Reply:
The idea of having this public is to allow using `g_object_new()` instead of forcing people to always use `cog_view_new()`. There is even an example that uses `g_object_new(cog_view_get_impl_type(), ...)` in the [documentation for cog_view_get_impl_type](https://igalia.github.io/cog/type_func.View.get_impl_type.html).

If we are allowing instantiation of the platform-provided custom `CogView` subclasses via `cog_view_new()`, I think there is no problem at all in allowing to use `g_object_new()` in combination with this function in the public API.

Reply:
I see then I don't have any particular issue with this.